test/parallel/test-fs-append-file.js

```diff
@@ -129,18 +129,34 @@ const throwNextTick = (e) => { process.nextTick(() => { throw e; }); };
     .catch(throwNextTick);
 }
 
-// Test that appendFile does not accept numbers (callback API).
+// Test that appendFile does not accept invalid data type (callback API).
 [false, 5, {}, [], null, undefined].forEach((data) => {
   const errObj = {
     code: 'ERR_INVALID_ARG_TYPE',
     message: /"data"|"buffer"/
   };
+  const filename = join(tmpdir.path, 'append-invalid-data.txt');
+
+  assert.throws(
+    () => fs.appendFile(filename, data, common.mustNotCall()),
+    errObj
+  );
+
   assert.throws(
-    () => fs.appendFile('foobar', data, common.mustNotCall()),
+    () => fs.appendFileSync(filename, data),
+    errObj
+  );
+
+  assert.rejects(
+    fs.promises.appendFile(filename, data).finally(() => {
```

Comment:
Instead of using finally here, I suggest to await `assert.rejects` and to verify that the file does not exist afterwards.